hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/DestroyProxyMessageTask.java

```diff
@@ -18,29 +18,57 @@
 
 import com.hazelcast.client.impl.protocol.ClientMessage;
 import com.hazelcast.client.impl.protocol.codec.ClientDestroyProxyCodec;
+import com.hazelcast.cluster.Member;
+import com.hazelcast.core.HazelcastInstanceNotActiveException;
 import com.hazelcast.instance.impl.Node;
 import com.hazelcast.internal.nio.Connection;
 import com.hazelcast.security.permission.ActionConstants;
-import com.hazelcast.spi.impl.proxyservice.ProxyService;
+import com.hazelcast.spi.impl.operationservice.Operation;
+import com.hazelcast.spi.impl.proxyservice.impl.operations.DistributedObjectDestroyOperation;
 
 import java.security.Permission;
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.logging.Level;
 
+import static com.hazelcast.internal.util.ExceptionUtil.peel;
 import static com.hazelcast.security.permission.ActionConstants.getPermission;
+import static java.util.logging.Level.FINEST;
+import static java.util.logging.Level.WARNING;
 
-public class DestroyProxyMessageTask extends AbstractCallableMessageTask<ClientDestroyProxyCodec.RequestParameters>
-        implements BlockingMessageTask {
+public class DestroyProxyMessageTask extends AbstractMultiTargetMessageTask<ClientDestroyProxyCodec.RequestParameters> {
 
     public DestroyProxyMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
         super(clientMessage, node, connection);
     }
 
     @Override
-    protected Object call() throws Exception {
-        ProxyService proxyService = nodeEngine.getProxyService();
-        proxyService.destroyDistributedObject(parameters.serviceName, parameters.name);
+    protected Supplier<Operation> createOperationSupplier() {
+        return () -> new DistributedObjectDestroyOperation(parameters.serviceName, parameters.name);
```

Comment:
We can make `DestroyProxyMessageTask` implement the Supplier, so that this method will no create a lamda but return `this` instead.